mypy/server/aststrip.py

```diff
@@ -233,8 +233,8 @@ def visit_import_all(self, node: ImportAll) -> None:
         node.imported_names = []
 
     def visit_for_stmt(self, node: ForStmt) -> None:
-        node.index_type = None
```

Comment:
This is not correct, since we mutate `index_type` in the semantic analyzer. Instead, you can add a new attribute called `unanalyzed_index_type` that contains the original type from the parser, and then we can keep this assignment here. This is similar to how we deal with annotations in `AssignmentStmt` nodes.

Reply:
Fixed